src/test/java/com/iggroup/universityworkshopmw/integration/OpenPositionFlowIntegrationTest.java

```diff
@@ -0,0 +1,70 @@
+package com.iggroup.universityworkshopmw.integration;
+
+import com.iggroup.universityworkshopmw.domain.model.Client;
+import com.iggroup.universityworkshopmw.domain.services.ClientService;
+import com.iggroup.universityworkshopmw.domain.services.OpenPositionsService;
+import com.iggroup.universityworkshopmw.integration.controllers.OpenPositionsController;
+import com.iggroup.universityworkshopmw.integration.dto.OpenPositionDto;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.test.web.servlet.MockMvc;
+
+import static com.iggroup.universityworkshopmw.TestHelper.APPLICATION_JSON_UTF8;
+import static com.iggroup.universityworkshopmw.TestHelper.convertObjectToJsonBytes;
+import static com.iggroup.universityworkshopmw.integration.transformers.OpenPositionTransformer.transform;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.springframework.test.web.servlet.setup.MockMvcBuilders.standaloneSetup;
+
+public class OpenPositionFlowIntegrationTest {
+
+   private MockMvc mockMvc;
+
+   private OpenPositionsController openPositionsController;
+   private ClientService clientService = spy(ClientService.class);
+   private OpenPositionsService openPositionsService = spy(new OpenPositionsService(clientService));
+
+   @Before
+   public void setup() {
+      openPositionsController = new OpenPositionsController(openPositionsService);
+      mockMvc = standaloneSetup(openPositionsController).build();
+   }
+
+   @Test
+   public void openPositionFlow() throws Exception {
+      Client client = clientService.storeNewClient(createClient());
+      OpenPositionDto openPositionDto = createOpenPositionDto();
+      mockMvc.perform(post("/openPositions/" + client.getId())
+         .contentType(APPLICATION_JSON_UTF8)
+         .content(convertObjectToJsonBytes(openPositionDto))
+      )
+         .andExpect(status().isOk())
+         .andExpect(content().contentType(APPLICATION_JSON_UTF8))
+         .andReturn();
+
+      verify(openPositionsService, times(1)).addOpenPositionForClient(client.getId(), transform(openPositionDto));
+      verify(clientService, times(1)).updateFunds(client.getId(), 10000.00);
+   }
```

Comment:
need to do test for adding position and deleting position - after merging this PR we can work on fixing the tests so that in controller tests we call controller methods and not mock mvc as we dicussed :)